src/pybind/mgr/alert/module.py

```diff
@@ -0,0 +1,194 @@
+
+"""
+An email alert module to report ceph cluster health status
+
+See doc/mgr/EmailReport.rst for more info.
+"""
+
+from mgr_module import MgrModule
+from threading import Event
+import sys
+import  six
+import smtplib
+import json
+import datetime
+from email.mime.text import MIMEText
+import Slacker
+import tweepy
+from config import CONFIG
+
+SLEEP_INTERVAL = datetime.timedelta(seconds = 60)
+
+class Alert(MgrModule):
+    OPTIONS = [
+    ''' define the servers address and some authentication credentials'''
+            {
+                'name': 'mail_server',
+                'default': 'smtp.gmail.com'
+            },
+            {
+                'name': 'mail_port',
+                'default': 587
+            },
+            {
+                'name': 'email_receiver',
+                'default': None
+            },
+            {
+                'name': 'email_sender',
+                'default': None
+            },
+            {
+                'name': 'sender_password',
+                'default': None
+            },
+            {
+                'name': 'slack_api',
+                'default': None
+            },
+            {
+                'name': 'twitter_consumer_key'
+                'default': None
+            },
+            {
+                'name': 'twitter_consumer_secret'
+                'default': None
+            },
+            {
+                'name': 'twitter_access_token'
+                'default': None
+            },
+            {
+                'name': 'twitter_access_token_secret'
+                'default': None
+            }
+    ]
+
+    COMMANDS = [
+        {
+            "cmd": "email alert "
+                   "name=status_code,type=CephString,req=false",
+            "desc": "Send an email alert to inform user of their Ceph cluster status",
+            "perm": "r"
+        },
+        {
+            "cmd": "slack alert ",
+            "desc": "post to slack",
+            "perm": "r"
+        }
+    ]
+
+    def __init__(self, *args, **kwargs):
+        super(Module, self).__init__(*args, **kwargs)
+        # set up some members to enable the serve() method and shutdown
+        self.run = True
+        self.event = Event()
+
+    def init_module_config(self):
```

Comment:
i think you can drop this whole function and just call self.notify_config() from __init__

Reply:
That puts the config options in a member variable of self.  for example, ``self.twitter_access_token`` will be None unless the config value is set.

Reply:
Ok. I am still looking up on this. I didn't know about this